google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -1145,15 +1152,31 @@ public Database call() throws Exception {
 
   @Override
   public OperationFuture<Backup, CreateBackupMetadata> createBackup(
-      final String instanceName, final String backupId, final Backup backup)
-      throws SpannerException {
-    CreateBackupRequest request =
+      final com.google.cloud.spanner.Backup backupInfo) throws SpannerException {
+    final String instanceName = backupInfo.getInstanceId().getName();
+    final String databaseName = backupInfo.getDatabase().getName();
+    final String backupId = backupInfo.getId().getBackup();
+    final Backup.Builder backupBuilder =
+        com.google.spanner.admin.database.v1.Backup.newBuilder()
+            .setDatabase(databaseName)
+            .setExpireTime(backupInfo.getExpireTime().toProto());
```

Comment:
I don't think it is guaranteed that `backupInfo.getExpireTime()` will always be set, so this could cause a `NullPointerException`.

Reply:
We shouldn't be able to create a backup without an expire time (https://github.com/googleapis/java-spanner/blob/32fdd606f392bc97dab7f37b1c566b3954839f7e/google-cloud-spanner/src/main/java/com/google/cloud/spanner/Backup.java#L64-L65). I moved these checks into the database client instead (with illegal argument exception, instead of illegal state exceptions), so that we prevent the creation of a Backup without the necessary parameters.